packages/rpc-core/src/rpc-methods/__tests__/get-genesis-hash-test.ts

```diff
@@ -0,0 +1,32 @@
+import { createHttpTransport, createJsonRpc } from '@solana/rpc-transport';
+import type { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import fs from 'fs';
+import fetchMock from 'jest-fetch-mock-fork';
+import path from 'path';
+
+import { createSolanaRpcApi, SolanaRpcMethods } from '../index';
+
+const logFilePath = path.resolve(__dirname, '../../../../../test-ledger/validator.log');
+const genesisHashPattern = /genesis hash: ([\d\w]{44})/;
+
+describe('getGenesisHash', () => {
+    let rpc: Rpc<SolanaRpcMethods>;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        rpc = createJsonRpc<SolanaRpcMethods>({
+            api: createSolanaRpcApi(),
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+    });
+
+    describe('when sent to a local validator', () => {
+        it('returns the genesis hash', async () => {
+            expect.assertions(1);
+            const logFile = fs.readFileSync(logFilePath, 'utf-8');
```

Comment:
Ooh. Interesting.

One downside of this is that it will read the whole log into memory. The later this test runs, the larger that logfile is.

Alternatively, you could read lines and read _until_ you find the genesis hash.

```ts
import { open } from 'node:fs/promises';

const file = await open('./some/file/to/read');

let expectedGenesisHash;
for await (const line of file.readLines()) {
  const match = line.match(genesisHashPattern);
  if (match) {
    expectedGenesisHash = match[1];
    break;
  }
}
if (expectedGenesisHash === undefined) {
  throw new Error('...');
}
```

Reply:
Hey! @mcintyre94 actually re-wrote this to use chunked reads for this reason! #1521

Reply:
Hmm I guess we could trim it down to lines though if we want to! Nice touch

Reply:
Yep we hit this! https://github.com/solana-labs/solana-web3.js/pull/1521/files
I used a stream and it’s a bit more code than your example there, but similar idea

Reply:
#1535